pkgs/by-name/rb/rbw/package.nix

```diff
@@ -54,6 +51,27 @@ rustPlatform.buildRustPackage rec {
       --fish <($out/bin/rbw gen-completions fish) \
       --zsh <($out/bin/rbw gen-completions zsh)
   ''
+  + lib.optionalString (stdenv.buildPlatform.canExecute stdenv.hostPlatform) ''
+    for gen in nushell powershell elvish fig; do
+      case "$gen" in
+        nushell)    dir=share/nushell/completions;    file=rbw.nu ;;
+        powershell) dir=share/powershell/completions; file=rbw.ps1 ;;
+        elvish)     dir=share/elvish/completions;     file=rbw.elv ;;
+        fig)        dir=share/fig/completions;        file=rbw.ts ;;
+      esac
+
+      echo "rbw: generating $gen completion..." >&2
+      payload="$($out/bin/rbw gen-completions "$gen" 2>/dev/null || true)"
+
+      if test -n "$payload"; then
+        install -d "$out/$dir"
+        printf '%s\n' "$payload" > "$out/$dir/$file"
+        echo "rbw: wrote $gen completion to $dir/$file" >&2
+      else
+        echo "rbw: $gen completion not supported; skipping" >&2
```

Comment:
To me, this should be a build error, rather than a warning (and lead us to remove the unsupported shell from the list).